avocado/plugins/run.py

```diff
@@ -196,4 +196,5 @@ def run(self, args):
             if len(job_instance.result_proxy.output_plugins) > 0:
                 result = job_instance.result_proxy.output_plugins[0]
                 result_dispatcher.map_method('render', result, job_instance)
+        job_instance.post_run()
```

Comment:
I don't like the non-parity of the `pre` and `post` job hooks. Actually the more I look at this section, the more I think that the `result_dispatcher` should not be triggered inside `plugins.run`, but it should be core part of the `job`.

All in all I like the idea of `post_run` function, but I think it should be defined as internal `job` function and it should run at the very end in the order which is desired and described in documentation.